prov/efa/src/rxr/rxr_init.c

```diff
@@ -200,7 +202,13 @@ int rxr_raw_addr_to_smr_name(void *ptr, char *smr_name)
 		return -errno;
 	}
 
-	ret = snprintf(smr_name, SMR_NAME_MAX, "%s_%04x_%08x_%04x",
+	*smr_name = malloc(SMR_NAME_BUFF_SIZE);
```

Comment:
I'd prefer that the caller of this does the malloc instead, since it's calling free.

Reply:
The caller will need a size. I will introduce an internal macro EFA_SHM_MAX_NAME, and set it to 256.

Reply:
That seems reasonable to me, thanks.

Reply:
Done in new revision.